spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala

```diff
@@ -169,84 +176,93 @@ trait StandardizationExecution extends CommonJobExecution {
-                                                fsUtils: FileSystemVersionUtils): DataFrame = {
+                                                fsUtils: DistributedFsUtils): DataFrame = {
     import za.co.absa.atum.AtumImplicits._
     val fieldRenames = SchemaUtils.getRenamesInSchema(schema)
     fieldRenames.foreach {
       case (destinationName, sourceName) => standardizedDF.registerColumnRename(sourceName, destinationName)
     }
 
-    // standardizedDF.setCheckpoint(s"$sourceId - End", persistInDatabase = false)  // TODO fix for s3 [ref issue #1416]
+    standardizedDF.setCheckpoint(s"$sourceId - End", persistInDatabase = false)
 
-    // TODO fix for s3 [ref issue #1416]
-//    val recordCount = standardizedDF.lastCheckpointRowCount match {
-//      case None => standardizedDF.count
-//      case Some(p) => p
-//    }
-    val recordCount = standardizedDF.count()
+    val recordCount = standardizedDF.lastCheckpointRowCount match {
+      case None => standardizedDF.count
+      case Some(p) => p
+    }
 
     if (recordCount == 0) {
       handleEmptyOutput(sourceId)
     }
 
     log.info(s"Writing into standardized path ${preparationResult.pathCfg.standardizationPath}")
     standardizedDF.write.parquet(preparationResult.pathCfg.standardizationPath)
+
     // Store performance metrics
     // (record count, directory sizes, elapsed time, etc. to _INFO file metadata and performance file)
 
-    // TODO fix for s3 [ref issue #1416]
-    // val stdDirSize = fsUtils.getDirectorySize(preparationResult.pathCfg.standardizationPath)
-    // preparationResult.performance.finishMeasurement(stdDirSize, recordCount)
-//    PerformanceMetricTools.addPerformanceMetricsToAtumMetadata(
-//      spark,
-//      "std",
-//      preparationResult.pathCfg.rawPath,
-//      preparationResult.pathCfg.standardizationPath,
-//      menasCredentials.username,
-//      args.mkString(" ")
-//    )
-
-    // TODO fix for s3 [ref issue #1416]
-    //cmd.rowTag.foreach(rowTag => Atum.setAdditionalInfo("xml_row_tag" -> rowTag))
-    //cmd.csvDelimiter.foreach(delimiter => Atum.setAdditionalInfo("csv_delimiter" -> delimiter))
-
-    // standardizedDF.writeInfoFile(preparationResult.pathCfg.standardizationPath)  // TODO fix for s3 [ref issue #1416]
-    //writePerformanceMetrics(preparationResult.performance, cmd)
+    val stdDirSize = fsUtils.getDirectorySize(preparationResult.pathCfg.standardizationPath)
+    preparationResult.performance.finishMeasurement(stdDirSize, recordCount)
+
+    PerformanceMetricTools.addPerformanceMetricsToAtumMetadata(
+      spark,
+      "std",
+      preparationResult.pathCfg.rawPath,
+      preparationResult.pathCfg.standardizationPath,
+      menasCredentials.username,
+      args.mkString(" ")
+    )
+
+    cmd.rowTag.foreach(rowTag => Atum.setAdditionalInfo("xml_row_tag" -> rowTag))
+    cmd.csvDelimiter.foreach(delimiter => Atum.setAdditionalInfo("csv_delimiter" -> delimiter))
```

Comment:
Why do we only provide these 2 as additional info? This might a mistake even in the master branch.

Reply:
This was in the code before, I haven't changed it, so I don't know.

Reply:
I wanted to tag @benedeki and somehow failed.

Reply:
I think the others are added, but these ones are added only if they are set

Reply:
these are the only reader options provided to additional info